### PR TITLE
19387 update getting SP entity role

### PR DESCRIPTION
**Pull request by kzdev420**
*Issue #:* /bcgov/entity#19387

*Description of changes:*

- FM1028594 - SP Individual
Screenshot 2024-02-29 at 8 14 55 AM

- FM1054864 - SP LEAR DBA
Screenshot 2024-02-29 at 8 14 22 AM

- FM1061225 - SP COLIN DBA
Screenshot 2024-02-29 at 8 13 44 AM

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the lear license (Apache 2.0).

### CODE REVIEW

File: legal-api/src/legal_api/resources/v2/business/business_parties.py [hunk 1]
```diff
@@ -43,8 +43,13 @@ def get_parties(identifier, party_id=None):
             HTTPStatus.UNAUTHORIZED,
         )
 
+    business_id = business.id
+    if business.is_owned_by_legal_entity_person or business.is_owned_by_legal_entity_org:
+        business_id = business.legal_entity_id
+    elif business.is_owned_by_colin_entity:
+        business_id = business.colin_entity_id
```
Comment (argush3) [inline legal-api/src/legal_api/resources/v2/business/business_parties.py:50]: so this won't work.

There are no `entity_role` entries for all the SP scenarios.

The code needs to be updated such that if `business.entity_type` == `SP`, party json is still returned.

Right now if you run this endpoint locally for the following SPs, nothing is returned right?
* FM1028594 - SP Individual
* FM1054864 - SP LEAR DBA
* FM1061225 - SP COLIN DBA

File: legal-api/src/legal_api/models/alternate_name.py [hunk 2]
```diff
@@ -182,6 +182,42 @@ def is_owned_by_colin_entity(self):
         """Return if owned by colin entity."""
         return bool(self.colin_entity)
 
+    @property
+    def owner_data_json(self):
+        """Return if owner data."""
+        json = {
+            "officer": {},
+            "roles": [{"appointmentDate": "2000-01-03", "cessationDate": None, "roleType": "Proprietor"}],
```
Comment (argush3) [inline legal-api/src/legal_api/models/alternate_name.py:190]: Populate `appointmentDate` with `self.startDate`

File: legal-api/src/legal_api/models/alternate_name.py [hunk 3]
```diff
@@ -182,6 +182,42 @@ def is_owned_by_colin_entity(self):
...
+                "partyType": "organization",
+            }
+        elif self.is_owned_by_colin_entity:
+            json["officer"] = {
+                "id": self.colin_entity.id,
+                "email": self.colin_entity.email,
+                "identifier": self.colin_entity.identifier,
+                "organizationName": self.legal_name,
+                "partyType": "organization",
+            }
```
Comment (argush3) [inline legal-api/src/legal_api/models/alternate_name.py:217]: Mailing and delivery address for the proprietor needs to be returned.  

No data has been populated yet, but please use addresses associated with `alternate_names.delivery_address_id` and `alternate_names.mailing_address_id`.  There will be updates to where addresses are pulled from in another ticket as I had mentioned before.  We just want to placeholder logic in place for addresses.

File: legal-api/src/legal_api/resources/v2/business/business_parties.py (hunk 1 above)
Comment (argush3) [reply legal-api/src/legal_api/resources/v2/business/business_parties.py]: Can you verify your code still works for other business types?

File: legal-api/src/legal_api/models/alternate_name.py (hunk 2 above)
Comment (kzdev420) [reply legal-api/src/legal_api/models/alternate_name.py]: fixed

Comment (argush3) [reply legal-api/src/legal_api/models/alternate_name.py]: Can you give me a screenshot what appointmentDate looks like via postman call?

File: legal-api/src/legal_api/models/alternate_name.py (hunk 3 above)
Comment (kzdev420) [reply legal-api/src/legal_api/models/alternate_name.py:221]: Updated

**Review by argush3 — APPROVED** [review]

**Pull request merged**